pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/openshift/installer/pkg/validate"
+	"gopkg.in/AlecAivazis/survey.v1"
+	"gopkg.in/AlecAivazis/survey.v1/terminal"
+)
+
+// Platform collects bare metal specific configuration.
+func Platform() (*baremetal.Platform, error) {
+	var provisioningNetworkCIDR, externalBridge, provisioningBridge, provisioningNetworkInterface string
+	var hosts []*baremetal.Host
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning Network CIDR",
+				Help:    "The network used for provisioning.",
+				Default: "172.22.0.0/24",
+			},
+			Validate: survey.ComposeValidators(survey.Required, ipNetValidator),
+		},
+	}, &provisioningNetworkCIDR); err != nil {
+		return nil, err
+	}
+	provNetCIDR, err := ipnet.ParseCIDR(provisioningNetworkCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "External bridge",
+				Help:    "External bridge is used for external communication by the bootstrap virtual machine.",
+				Default: baremetaldefaults.ExternalBridge,
+			},
+		},
+	}, &externalBridge); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning bridge",
+				Help:    "Provisioning bridge is used to provision machines by the bootstrap virtual machine.",
+				Default: baremetaldefaults.ProvisioningBridge,
+			},
+		},
+	}, &provisioningBridge); err != nil {
+		return nil, err
+	}
+
+	if err := survey.Ask([]*survey.Question{
+		{
+			Prompt: &survey.Input{
+				Message: "Provisioning Network Interface",
+				Help:    "The name of the network interface on a control plane host connected to the provisioning network.",
+			},
+			Validate: survey.Required,
+		},
+	}, &provisioningNetworkInterface); err != nil {
+		return nil, err
+	}
+
+	// Keep prompting for hosts
+	for {
+		var hostRole string
+		survey.AskOne(&survey.Select{
+			Message: "Add a Host:",
+			Options: []string{"control plane", "worker"},
+		}, &hostRole, nil)
+
+		var host *baremetal.Host
+		var err error
+		host, err = Host()
+		// Check for kebyoard interrupt or else we'll loop forever
+		if err == terminal.InterruptErr {
+			fmt.Println("interrupted - host not added")
+			break
+		} else if err != nil {
+			fmt.Printf("invalid host - please try again")
+			continue
+		}
+		hosts = append(hosts, host)
+
+		more := false
+		survey.AskOne(&survey.Confirm{
+			Message: "Add another host?",
+		}, &more, nil)
+		if !more {
+			break
+		}
+	}
+
+	return &baremetal.Platform{
+		ExternalBridge:               externalBridge,
+		ProvisioningBridge:           provisioningBridge,
+		ProvisioningNetworkCIDR:      provNetCIDR,
+		ProvisioningNetworkInterface: provisioningNetworkInterface,
+		Hosts:                        hosts,
+	}, nil
+}
+
+// uriValidator validates if the answer provided in prompt is a valid
+// url and has non-empty scheme.
+func uriValidator(ans interface{}) error {
```

Comment:
nit: move this to host.go as that's where it's being used...

Reply:
Done.